alts/src/test/java/io/grpc/alts/AuthorizationUtilTest.java

```diff
@@ -0,0 +1,114 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.alts;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import io.grpc.Attributes;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.grpc.ServerCall;
+import io.grpc.Status;
+import io.grpc.alts.internal.AltsAuthContext;
+import io.grpc.alts.internal.AltsProtocolNegotiator;
+import io.grpc.alts.internal.HandshakerResult;
+import io.grpc.alts.internal.Identity;
+import javax.annotation.Nullable;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for {@link AuthorizationUtil}. */
+@RunWith(JUnit4.class)
+public final class AuthorizationUtilTest {
+
+  @Test
+  public void altsAuthorizationCheck() throws Exception {
+    Status status =
+        AuthorizationUtil.clientAuthorizationCheck(
+            new FakeServerCall(null), ImmutableList.of("Alice"));
+    assertThat(status.getCode()).isEqualTo(Status.Code.NOT_FOUND);
+    assertThat(status.getDescription()).startsWith("Peer ALTS AuthContext not found");
+    status =
+        AuthorizationUtil.clientAuthorizationCheck(
+            new FakeServerCall("Alice"), ImmutableList.of("Alice", "Bob"));
+    assertThat(status.getCode()).isEqualTo(Status.Code.OK);
+    status =
+        AuthorizationUtil.clientAuthorizationCheck(
+            new FakeServerCall("Alice"), ImmutableList.of("Bob", "Joe"));
+    assertThat(status.getCode()).isEqualTo(Status.Code.PERMISSION_DENIED);
+    assertThat(status.getDescription()).endsWith("not authorized");
+  }
+
+  private static class FakeServerCall extends ServerCall<String, String> {
+    final Attributes attrs;
+
+    FakeServerCall(@Nullable String peerServiceAccount) {
+      Attributes.Builder attrsBuilder = Attributes.newBuilder();
+      if (peerServiceAccount != null) {
+        HandshakerResult handshakerResult =
+            HandshakerResult.newBuilder()
+                .setPeerIdentity(Identity.newBuilder().setServiceAccount(peerServiceAccount))
+                .build();
+        AltsAuthContext altsAuthContext = new AltsAuthContext(handshakerResult);
+        attrsBuilder.set(AltsProtocolNegotiator.AUTH_CONTEXT_KEY, altsAuthContext);
+      }
+      attrs = attrsBuilder.build();
+    }
+
+    @Override
+    public void request(int numMessages) {
+      throw new AssertionError("Should not be called");
```

Comment:
A trick you can do for these is to extend ForwardingServerCall and then implement `delegate()` to throw like you're doing here. This is fine, though.

Reply:
Ack on the trick. Let's keep the current FakeServerCall if you don't mind.